.github/workflows/version.yml

```diff
@@ -0,0 +1,72 @@
+name: Bump version
+
+on:
+  push:
+    tags:
+    - "v*"
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Get commits
+      id: commits
+      run: |
+        echo "MAIN=$(git show -s --format="%H" origin/main)" >> "$GITHUB_OUTPUT"
+        echo "TAG=$(git rev-list -n 1 "${GITHUB_REF#refs/*/}")" >> "$GITHUB_OUTPUT"
+    - name: Print commits
+      run: |
+        echo "Main commit: ${{ steps.commits.outputs.MAIN }}"
+        echo "Tag commit: ${{ steps.commits.outputs.TAG }}"
+    - name: Compare commits
+      if: "${{ steps.commits.outputs.MAIN != steps.commits.outputs.TAG }}"
+      uses: actions/github-script@d556feaca394842dc55e4734bf3bb9f685482fa0
+      with:
+        script: |
+            core.setFailed('Tagged commit does not match main')
+
+  bump-version:
+    runs-on: ubuntu-latest
+    needs: [validate]
+    env:
+      VERSION: "${{ github.ref_name }}"
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v3
+        with:
+          ref: "main"
+          ssh-key: "${{ secrets.SSH_KEY }}"
```

Comment:
The issue with this approach is that the `SSH_KEY` variable is just as critical as the PyPi token.

Reply:
Right. I'll add the PyPi environment here as well

Reply:
Hm second thought, I don't think it's a great approach to reuse an environment for multiple workflows because if the first PyPi environment use fails (in bumping the version), the `PyPi` env will show up as `failed` in the repo. We could set it up as a second environment (`Staging`or `pre-PyPi` maybe?) and have the `ssh-key` there.

The other option (mentioned offline) would be to use `workflow_dispatch` and manually trigger the initial workflow. This removes the need to `push -f` the tag. We'd still need a second environment for the `ssh-key`.

The deeper this goes, the more I'm leaning toward just keeping the version bump as a manual process. The automation benefit just seems too small IMO. Should we stick with leaving this as a manual process, I think adding the workflow linter and PyPi environment for the `pypi_token` are great additions in and of themselves.

Reply:
> The other option (mentioned offline) would be to use workflow_dispatch and manually trigger the initial workflow

This is kind of what we went for with the Solidity contracts.

> We'd still need a second environment for the `ssh-key`.

I don't think this is the case! If the tag is not protected, you would manually dispatch a "release" workflow, you may be able to add an input parameter where you could specify patch/minor/major, the workflow creates a new tag and pushes it without -f, using the normal github token that the workflow gets.

Manually running the version bump is definitely okay with me too.

Reply:
Great point with the gh token!